src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,173 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+
+package app
+
+import org.eclipse.jgit.blame.BlameGenerator
+import org.eclipse.jgit.blame.BlameResult
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.revwalk.RevSort
+import org.eclipse.jgit.treewalk.filter.*
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.util.Date
+
+/**
+ * Used to store and identify a code line information in a repo.
+ */
+class CodeLine(val rev: String, val file: String, val line: Int,
+               val authorName: String, val authorEmail: String,
+               val age: Int) {
+}
+
+/**
+ * Used to compute age of code lines in the repo.
+ */
+class CodeLongevity(repo_path: String) {
+    var repo_path: String = repo_path
+
+    var list: MutableList<CodeLine> = mutableListOf()
+
+    init {
+        println("READYY>>>>>>>>>>");
+        println("Repo: $repo_path")
+
+        val repo = FileRepository(repo_path)
+        val revWalk = RevWalk(repo);
+        val head = revWalk.parseCommit(repo.readOrigHead()); // master
+
+        computeExistingLines(repo, head)
+        computeDeletedLines(repo, revWalk, head)
+    }
+
+    fun test(email: String) {
+        var sum: Long = 0;
```

Comment:
I wonder if filter() would be way slower than a for loop. Don't know, but would be a cool test.